### ISSUE TITLE
VO often reads responses in incorrect order

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/greenhouse-effect#192]
**Test device**
MacBook Air (m1 chip)

**Operating System**
12.4

**Browser**
safari + VoiceOver

**Problem description**
For https://github.com/phetsims/qa/issues/819, VoiceOver doesn't read statements in the order listed in https://github.com/phetsims/qa/issues/819#issuecomment-1182529346. VO often reads the IR emitting statement before the IR redirecting statement. 

**Steps to reproduce**
These steps have consistently worked for me:
1. With VO on, tab to and then press the Start Sunlight button
2. Tab to the Concentration slider and wait for IR rays to start emitting from the surface
3. Press the Up Arrow key once

**Visuals**

https://user-images.githubusercontent.com/87318828/178819484-f6942f46-4533-48ce-8134-dd1360a83519.mov

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Greenhouse Effect‬
URL: https://phet-dev.colorado.edu/html/greenhouse-effect/1.1.0-dev.10/phet/greenhouse-effect_all_phet.html
Version: 1.1.0-dev.10 2022-07-08 17:41:59 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.5 Safari/605.1.15
Language: en-US
Window: 1440x704
Pixel Ratio: 2/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0 (1.0)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 256
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}
</details>

**Comment by Nancy-Salpepi** [comment]
I also see the wrong order uttered in 1.1.0-dev.9. Sorry I didn't catch it.

**Comment by arouinfar** [comment]
Thanks @Nancy-Salpepi 

This seems pedagogically problematic to me. There is a cause and effect relationship that we want users to pick up on. More greenhouse gases in atmosphere means that more infrared is redirected to (and thus absorbed by) the surface resulting in the surface warming and emitting more infrared.

**Comment by jessegreenberg** [comment]
Screen readers will speak alerts in whatever order they wish, but we hoped UtteranceQueue would feed content to them so that alerts are made in the order that we add them to the queue. In the sim we could probably fix this by adding a small delay in between each alert we send to the UtteranceQueue. But it would be better if a fix can be made in UtteranceQueue directly.

**Comment by jessegreenberg** [comment]
Just noting that performance on my old MacOS is to bad to run VoiceOver with this sim. VoiceOver only ever says "Busy" while the sim is running. But I am able to test alerts by disabling drawing the waves.
```js
    // Update the view when changes occur to the modelled waves.
    model.wavesChangedEmitter.addListener( () => {
      // wavesCanvasNode.invalidatePaint();
    } );
```

**Comment by jbphet** [comment]
Discussed in the 7/27/2022 meeting, and @arouinfar said that it may be a couple of weeks before the planned interviews occur, so it would be nice to address this one prior to the interviews.

**Comment by jessegreenberg** [comment]
I have a proposed fix for this in https://github.com/phetsims/utterance-queue/issues/88 but I am not able to test VoiceOver with my Mac. @Nancy-Salpepi would you mind trying the sim on master and seeing if this is improved?

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg things are much improved! If I move the slider and allow all the alerts to play, they are in the correct order. 

I only hear statements out of order if I move the slider in the middle of the statements being uttered. Here is an example:

https://user-images.githubusercontent.com/87318828/181523184-7cfdbd6c-23cc-4878-8ad7-768d8ac018c7.mov

@jessegreenberg I'm assuming this is because VO doesn't interrupt alerts??

**Comment by jessegreenberg** [comment]
Yea, thats right @Nancy-Salpepi. We might be able to make the "More infrared radiation redirecting back to surface" alert "assertive" so that it interrupts and cancels all previous alerts Ill try it out.

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg the "More/Less radiation redirection back to surface" statement now interrupts alerts! There is a noise that  occurs each time before that particular statement. Not sure how you felt about that. 

https://user-images.githubusercontent.com/87318828/182634510-a203ecac-c05f-4ba8-99e7-1963ed19a780.mov

**Comment by Nancy-Salpepi** [comment]
Things sound a bit odd when I switch to Date mode. The Redirecting statement comes before the description of the time period. For example:
* Year 2020, selected, radio button, 1 of 4
* More infrared radiation redirecting back to surface
* Now there are a large number of homes and factories in the observation window.

**Comment by jessegreenberg** [comment]
OK, thanks @Nancy-Salpepi - regarding the sound in https://github.com/phetsims/greenhouse-effect/issues/192#issuecomment-1204030825 I checked with @terracoda and apparently that is a sound that VO plays. Most likely they are trying to signify that something important just happened with an aria-live=assertive alert. There is no way to prevent it currently.

Regarding https://github.com/phetsims/greenhouse-effect/issues/192#issuecomment-1204040202, yes you are right... Similarly bad output is happening if you change the time period in date mode. I think there is a different implementation that could keep the "assertive" alert and still work for date mode. It might take an hour to do and would be a bit disruptive to the current code. @arouinfar can you please review the recording in https://github.com/phetsims/greenhouse-effect/issues/192#issuecomment-1204030825 and comment on if this is something we should do? If that recording is not acceptable we can change the current alert implementation, otherwise Ill revert my change in https://github.com/phetsims/greenhouse-effect/commit/102839a16d3c5208a6738187c27dfcb494afc892.

**Comment by jbphet** [comment]
We discussed this in today's design meeting, and it was suggested that we try removing obsolete messages from the utterance queue when possible.  @jessegreenberg did some real-time investigation and said that this is indeed a possibility.  He'll try it out and then we will re-review.

**Comment by jessegreenberg** [comment]
Clearing the utterance queue will not help with this. I verified that the alerts have been given to the screen reader before they can be cleared in this case.

Order of alerts currently:
![image](https://user-images.githubusercontent.com/6396244/183957884-5ce34915-44d8-40bb-8387-063e1135f141.png)

We want to hear all of these, it is just that they are coming through in an odd order. This will require changes to the alerter code.

**Comment by jessegreenberg** [comment]
OK, a change was made to the description implementation so that we should always hear the observation window scene description before "infrared radiation redirecting" descriptions. This should resolve the comment in https://github.com/phetsims/greenhouse-effect/issues/192#issuecomment-1204040202.

@Nancy-Salpepi can you please try this situation again and see if it is improved?

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg I still hear IR radiation description before scene description

https://user-images.githubusercontent.com/87318828/184130433-7bb53082-0a08-42ec-90dc-d7192f33a374.mov

**Comment by jessegreenberg** [comment]
Thanks @Nancy-Salpepi! I can hear the "beep" in your screen recording from the assertive alert so I think this is happening because VO reads the assertive alert and then reads the polite alerts without cancelling them. I am going to go ahead and revert my change in https://github.com/phetsims/greenhouse-effect/commit/102839a16d3c5208a6738187c27dfcb494afc892.

Alerts should come out in the right order. But it is possible that https://github.com/phetsims/greenhouse-effect/issues/192#issuecomment-1204030825 could still happen.

@Nancy-Salpepi can you please confirm that your previous comment does not happen?

**Comment by Nancy-Salpepi** [comment]
@jessegreenberg the alerts are now coming out in the right order 😁

**Comment by jessegreenberg** [comment]
Hooray! Thanks @Nancy-Salpepi. Closing.
